### DISCUSSION TITLE
support for browsing and copying from USB flash drive

### DISCUSSION THREAD

**Discussion by kapaka** [koreader/koreader#10075] [Ideas]
Would it be possible to implement to copy files from a external USB flash drive connected to the micro USB port? 
Lately I have been reading manga in my Kobo Aura One. The limited internal capacity is not enough for all when traveling. Never had the problem with books as they are much smaller size.
Thanks :)

**Comment by NiLuJe** [comment]
That would involve USB OTG (which we do support for external input devices **on supported devices**: yours is not one of those), which is a first hurdle...

The second would be mounting the drive, which would basically only work for ext* & FAT* formatted drives, which is probably going to be a no-go for most non-Linux users (e.g., macOS, Windows or exFAT filesystems are unsupported).

The third hurdle may be power related on some and/or all devices, as I'm not sure enough power is supplied by the board for such use cases.

TL;DR: Nope, not gonna happen.

**Reply by Frenzie** [reply]
> The third hurdle may be power related on some and/or all devices, as I'm not sure enough power is supplied by the board for such use cases.

You can always use a powered hub. But yes, I imagine your average ereader would either fail to supply sufficient power or drain rather quickly.

**Reply by kapaka** [reply]
Thank you for the explanation about the impossibility of the concept .
Can you think of an alternate way to add files when traveling without a computer? Somehow over WiFi from Dropbox? Android phone? Can you point me to a explanation?
 Thank You

**Reply by mergen3107** [reply]
WebDav - use the Cloud storage option in KOReader’s File Manager (top menu). 

One good example is Yandex Disk via Webdav. It gives you 10 GB when you open an account.
